packages/did-core-test-server/suites/did-spec/did-metadata-structure.js

```diff
@@ -0,0 +1,61 @@
+const didMetadataStructureTest_r = (didDocumentMetadata) => {
+    describe("didDocumentMetadata", () => {
+        it('The structure used to communicate this metadata MUST be a map of properties.', () => {
+            expect(didDocumentMetadata instanceof Object).toBeTruthy();
+        });
+
+        it('Each property name MUST be a string.', () => {
+            Object.keys(didDocumentMetadata).forEach((property_name) => {
+                expect(typeof property_name).toBe("string");
+            });
+        });
+
+        it('Each property value MUST be a string, map, list, ordered set, boolean, or null.', () => {
+            Object.keys(didDocumentMetadata).forEach((property_name) => {
+                expect(typeof didDocumentMetadata[property_name]).toEqual(expect.stringMatching(/(string|object|boolean)/));  // map, list, set and null are objects
```

Comment:
a bit of a nit, but potentially we can test "list / set" better

Reply:
Any suggestion welcome. I think we need to have some is-a functions for all of the basic types and consistently use it.